### PR TITLE
JP-3523: Master background for IFU data needs to cover same wavelength range as the science data

### LINKED ISSUE #8244: Master background for IFU data  needs to cover same wavelength range as the science data
_Issue [JP-3523](https://jira.stsci.edu/browse/JP-3523) was created on JIRA by [Jane Morrison](https://jira.stsci.edu/secure/ViewProfile.jspa?name=morrison):_

PID 1717 showed a problem with master background subtraction. In the data the background observation failed which is the underlying reason the data failed to process. However the master background step should be "smart" enough to know that the background did not contain the full wavelength range of the science data.  In this particular case the master background did not contain CH 1 A or CH 1 B. The science data did contain these bands. When the master background was subtracted for these bands - nans were subtracted resulting in no valid data on the detector when cube_build later ran.
A plot of the master background used on the data is attached to this ticket.

After subtracting this master background the image of the data is also attached to this ticket showing the NAN regions plotted in green.

This ticket has been opened to discuss what should be done for the master background step logic when the full wavelength range in the science data is not covered in the background images.

### PR DESCRIPTION
**Pull request by penaguerrero**
Resolves [JP-3523](https://jira.stsci.edu/browse/JP-3523)

Closes #8244 

This PR adds changes suggested by D. Law in Jira ticket: "First interpolate the 1d master background vector to the wavelengths of the 2d science data like we're doing now.  Then look for all input science pixels whose wavelengths are more than (say) 0.1 microns away from the closest wavelength in the 1d master background vector, and set the 2d background for those pixels to zero."

**Checklist for PR authors (skip items if you don't have permissions or they are not applicable)**
- [x] added entry in `CHANGES.rst` within the relevant release section
- [ ] updated or added relevant tests
- [ ] updated relevant documentation
- [ ] added relevant milestone
- [x] added relevant label(s)
- [ ] ran regression tests, post a link to the Jenkins job below.
      [How to run regression tests on a PR](https://github.com/spacetelescope/jwst/wiki/Running-Regression-Tests-Against-PR-Branches)
- [ ] All comments are resolved
- [ ] Make sure the JIRA ticket is [resolved properly](https://github.com/spacetelescope/jwst/wiki/How-to-resolve-JIRA-issues)

### CODE REVIEW

**Comment by drlaw1558** [comment]
I think we need to take a different approach to this, and can do so much more simply by removing code instead of actually adding anything.  Right now there are a couple problems with the new code: the mask_limit calls around line 308 look to be the primary thing that has been causing us problems, and since those are still there we're still effectively nan-ing out science data.  At the same time, the runtime of the step has increased from 20 seconds to about 4 minutes.

Digging into it a little, I think we can achieve what we want scientifically in a much simpler way.  The only thing that this PR needs to do is remove lines 308-312 involving mask_limit, no new code required.  np.interp in line 314 looks like it already does what we want by setting left=0 and right=0, which ensures that the background would be zero-valued outside where we have wavelength coverage from the background spectra.

There's still the issue of missing background values within the background wavelength vector (e.g., if we were missing background values from just 4A), but that's not nearly as big a problem- the existing interpolating already does a pretty decent job inside vector.

So, with apologies for all the back and forth on this converging on the best solution, I think a very simple PR just deleting lines 305-311 of the existing code would do what we want, with even a very slight runtime improvement.

**Comment by penaguerrero** [comment]
running reg tests at: https://plwishmaster.stsci.edu:8081/job/RT/job/JWST-Developers-Pull-Requests/1544/

File: CHANGES.rst [hunk 1]
```diff
@@ -165,6 +165,9 @@ lib
 master_background
 -----------------
 
+- For MIRI_MRS, the background is be zero-valued outside the
```
Comment (hbushouse) [inline CHANGES.rst:168]: need to clean up the "is be" language

**Comment by hbushouse** [comment]
> I think we need to take a different approach to this, and can do so much more simply by removing code instead of actually adding anything. Right now there are a couple problems with the new code: the mask_limit calls around line 308 look to be the primary thing that has been causing us problems, and since those are still there we're still effectively nan-ing out science data. At the same time, the runtime of the step has increased from 20 seconds to about 4 minutes.
> 
> Digging into it a little, I think we can achieve what we want scientifically in a much simpler way. The only thing that this PR needs to do is remove lines 308-312 involving mask_limit, no new code required. np.interp in line 314 looks like it already does what we want by setting left=0 and right=0, which ensures that the background would be zero-valued outside where we have wavelength coverage from the background spectra.
> 
> There's still the issue of missing background values within the background wavelength vector (e.g., if we were missing background values from just 4A), but that's not nearly as big a problem- the existing interpolating already does a pretty decent job inside vector.
> 
> So, with apologies for all the back and forth on this converging on the best solution, I think a very simple PR just deleting lines 305-311 of the existing code would do what we want, with even a very slight runtime improvement.

@drlaw1558 Lines 305-311 in which code module? Those lines in expand_to_2d.py are just part of the doc string.

**Comment by drlaw1558** [comment]
@hbushouse Too many branches, at this point the easiest thing to do is to point to https://github.com/spacetelescope/jwst/pull/8597 which implements what I meant in the above comment.  I.e., only subtracting code, not adding anything.

**Comment by hbushouse** [comment]
Closing in favor of #8597

**Pull request closed without merging**